### PR TITLE
upgrade crd to v1

### PR DESCRIPTION
**Pull request by zhiweiyin318**
1. upgrade crd to v1
2. delete denynamespace webhook
3. fix conversion-gen path issue
story: https://github.com/open-cluster-management/backlog/issues/10443

### CODE REVIEW

File: deploy/foundation/hub/resources/crds/cluster.open-cluster-management.io_mirroredmanagedclusters.yaml [hunk 1]
```diff
@@ -1,208 +1,172 @@
 
 ---
-apiVersion: apiextensions.k8s.io/v1beta1
+apiVersion: apiextensions.k8s.io/v1
```
Comment (qiujian16) [inline deploy/foundation/hub/resources/crds/cluster.open-cluster-management.io_mirroredmanagedclusters.yaml:3]: I think we can delete this crd also, no one is using it.

File: cmd/webhook/app/server.go [hunk 2]
```diff
@@ -95,7 +88,7 @@ func (a *admissionHandler) serve(w io.Writer, r *http.Request, admit admitFunc)
 func (a *admissionHandler) mutateResource(ar v1beta1.AdmissionReview) *v1beta1.AdmissionResponse {
```
Comment (qiujian16) [inline cmd/webhook/app/server.go:88]: change to v1.AdmissioinReview

File: cmd/webhook/app/server.go [hunk 3]
```diff
@@ -95,7 +88,7 @@ func (a *admissionHandler) serve(w io.Writer, r *http.Request, admit admitFunc)
...
 	klog.V(2).Info("mutating custom resource")
 	raw := ar.Request.Object.Raw
-	crd := apiextensionsv1beta1.CustomResourceDefinition{}
+	crd := apiextensionsv1.CustomResourceDefinition{}
```
Comment (qiujian16) [inline cmd/webhook/app/server.go:91]: I am not sure why we use CustomResourceDefinition here. It seems to be an old bug.

File: pkg/apis/inventory/v1alpha1/baremetalasset_types.go [hunk 4]
```diff
@@ -59,6 +59,7 @@ type BareMetalAssetSpec struct {
 	Role Role `json:"role,omitempty"`
 
 	// ClusterDeployment which the asset belongs to.
+	// +kubebuilder:pruning:PreserveUnknownFields
```
Comment (qiujian16) [inline pkg/apis/inventory/v1alpha1/baremetalasset_types.go:62]: this seems to be weird, i need to check whether we do not use ObjectReference.

File: deploy/foundation/hub/resources/crds/cluster.open-cluster-management.io_mirroredmanagedclusters.yaml (hunk 1 above)
Comment (zhiweiyin318) [reply deploy/foundation/hub/resources/crds/cluster.open-cluster-management.io_mirroredmanagedclusters.yaml]: deleted

File: cmd/webhook/app/server.go (hunk 2 above)
Comment (zhiweiyin318) [reply cmd/webhook/app/server.go]: done

File: cmd/webhook/app/server.go (hunk 3 above)
Comment (qiujian16) [reply cmd/webhook/app/server.go:91]: let's update this part as a followup

File: pkg/apis/inventory/v1alpha1/baremetalasset_types.go (hunk 4 above)
Comment (qiujian16) [reply pkg/apis/inventory/v1alpha1/baremetalasset_types.go:62]: Let's update this as a followup

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**